tests/pytest/test_short_read.py

```diff
@@ -0,0 +1,524 @@
+            env.assertNotEqual(fullfilePath, None, message='testShortReadJson')
+            sendShortReads(env, fullfilePath)
+
+
+def sendShortReads(env, rdb_file):
+
+    # Add some initial content (keys) to test backup/restore/discard when short read fails
+    env.assertCmdOk('replicaof', 'no', 'one')
+    env.flush()
+    add_keys(env, 10, 'backup:key')
+    env.assertIsNotNone(env.cmd('json.get', 'backup:key:1', '$'))
+    env.assertIsNotNone(env.cmd('json.get', 'backup:key:10', '$'))
+
+    with open(rdb_file, mode='rb') as f:
+        full_rdb = f.read()
+    total_len = len(full_rdb)
+
+    env.assertGreater(total_len, SHORT_READ_BYTES_DELTA)
+    r = range(0, total_len + 1, SHORT_READ_BYTES_DELTA)
+    if (total_len % SHORT_READ_BYTES_DELTA) != 0:
+        r = r + range(total_len, total_len + 1)
+    for b in r:
+        rdb = full_rdb[0:b]
+        runShortRead(env, rdb, total_len)
+
+
+@Debug(False)
+def runShortRead(env, data, total_len):
+    with ShardMock(env) as shardMock:
+
+        # For debugging: if adding breakpoints in redis,
+        # In order to avoid closing the connection, uncomment the following line
+        # res = env.cmd('CONFIG', 'SET', 'timeout', '0')
+
+        # Notice: Do not use env.expect in this test
+        # (since it is sending commands to redis and in this test we need to follow strict hand-shaking)
+        res = env.cmd('CONFIG', 'SET', 'repl-diskless-load', 'swapdb')
+        env.assertTrue(res)
+        res = env.cmd('replicaof', 'localhost', shardMock.server_port)
+        env.assertTrue(res)
+        conn = shardMock.GetConnection()
+        # Perform hand-shake with replica
+        res = conn.read_request()
+        env.assertEqual(res, ['PING'])
+        conn.send_status('PONG')
+
+        max_attempt = 100
+        res = conn.read_request()
+        while max_attempt > 0:
+            if res[0] == 'REPLCONF':
+                conn.send_status('OK')
+            else:
+                break
+            max_attempt = max_attempt - 1
+            res = conn.read_request()
+
+        # Send RDB to replica
+        some_guid = 'c43cf134fd16468b1e26a3d000f2053fef1c5f8c'
+        conn.send_status('FULLRESYNC ' + some_guid + ' 0')
+        is_shortread = total_len != len(data)
+        if is_shortread:
+            # Send without the trailing '\r\n' (send data not according to RESP protocol)
+            binary_data = b'$%d\r\n%s' % (total_len, data)
+            conn.send(binary_data)
+        else:
+            # Allow to succeed with a full read (send data according to RESP protocol)
+            conn.send_bulk(data)
+        conn.flush()
+
+        # Close during replica is waiting for more RDB data (so replica will re-connect to master)
+        conn.close()
+
+        # Make sure replica did not crash
+        res = env.cmd('PING')
+        env.assertEqual(res, True)
+        conn = shardMock.GetConnection(timeout=3)
+        env.assertNotEqual(conn, None)
+
+        if is_shortread:
```

Comment:
It looks like you are checking the same thing on the if and else, am I missing something?